client/packages/host/src/components/Initialise/hooks.ts

```diff
@@ -1,144 +1,185 @@
+  setSiteCredentialsError: error => set(state => ({ ...state, error })),
+  setIsLoading: isLoading => set(state => ({ ...state, isLoading })),
+  setPassword: password => set(state => ({ ...state, password })),
+  setUsername: username => set(state => ({ ...state, username })),
+  setUrl: url => set(state => ({ ...state, url })),
+  // When sync is already ongoing either after initialise button is pressed
+  // or when initialisation page is loaded while sync is ongoing
+  // inputs should be disabled and polling for syncStatus should start
+  setIsInitialising: isInitialising =>
+    set(state => ({
+      ...state,
+      isInitialising,
+      refetchInterval: isInitialising && POLLING_INTERVAL,
+      password: '',
+    })),
 }));
 
+// Hook will navigate to login if SyncState is Initialised
 export const useInitialiseForm = () => {
   const state = useInitialiseFormState();
   const navigate = useNavigate();
-  const { mutateAsync: restart } = useHost.utils.restart();
   const {
     setIsLoading,
     password,
-    setPassword,
     username,
-    error,
-    setError,
+    setSiteCredentialsError,
     url,
+    refetchInterval,
+    setIsInitialising,
+    setUrl,
+    setUsername,
   } = state;
-  const [isPolling, setIsPolling] = useState(false);
-  const [isBootstrap, setIsBootstrap] = useState(false);
-  const { mutateAsync: update } = useHost.sync.update();
-  const { data } = useHost.utils.settings({
-    refetchInterval: POLLING_INTERVAL,
-    enabled: isPolling,
-  });
   const t = useTranslation('app');
-  const parseErrorMessage = (error: Error, defaultKey: LocaleKey) => {
-    const matches = /code: "([a-zA-Z_]+?)"/g.exec(error?.message);
-    const key =
-      matches && matches.length > 1
-        ? (`error.${matches[1]}` as LocaleKey)
-        : defaultKey;
+  const { mutateAsync: initialise } = useHost.sync.initialise();
+  const { mutateAsync: manualSync } = useHost.sync.manualSync();
+  // Both syncState and syncStatus are polled because we want to navigate
+  // to login when initialisation is finished, but syncStatus will be behind auth after
+  // initialisation has finished, whereas syncStatus is always an open API
+  const { data: syncState } = useHost.utils.syncState(refetchInterval);
+  const { data: syncStatus } = useHost.utils.syncStatus(refetchInterval);
+  const { data: syncSettings } = useHost.utils.syncSettings();
 
-    return t(key);
-  };
-
-  const onSave = async () => {
-    setError();
+  const onInitialise = async () => {
+    setSiteCredentialsError();
     setIsLoading(true);
-    setIsBootstrap(false);
     const syncSettings = {
-      intervalSec: 300,
+      intervalSec: SYNC_INTERVAL,
       password,
       url,
       username,
     };
     try {
-      await update(syncSettings);
+      await initialise(syncSettings);
+      setIsInitialising(true);
     } catch (e) {
-      setError({
-        message: parseErrorMessage(e as Error, 'error.unable_to_save_settings'),
+      console.log(e);
```

Comment:
`console.error` 🤷

Reply:
Ohh, even if you catch the error it is logged ?

Reply:
that's because we have two layers of packages.. there's react query which is what we're dealing with here, but then beneath that there is a graphQL transport layer package which actually calls the API. That one we've extended slightly, to recognise errors, log them and throw an error which react query can then respond to.

The issue was that the graphQL api is correctly returning a 200 response with an error object - but the gql layer we are using sees the 200 and decides that it's a valid response and wouldn't throw an error. We're now interrogating the response, so that we can throw an error.

I'm sure it could be neater - but is working for now..

Reply:
In upstream PR i attempt to do something different, I'll keep it in mind 🙏